### ISSUE TITLE
(opensearch): Opensearch Construct creates the CNAME in a wrong way.

### ISSUE THREAD

**Issue by rix0rrr** [aws/aws-cdk#17681]
### What is the problem?

(Ticket came in via internal request, ref P55583243)

in 'monocdk' aws-opensearchservice constructsm the construct Domain has a bug.
given I have a private zone:

```
          const someZone = new PrivateHostedZone(this, 'OSStageHostedZone', {
       zoneName: 'domain.name.com',
       vpc: someVpc,
     };
```

when we provide customEndpoint configurations as the following for the Domain Constuct:

```
{
         domainName: 'elasticsearch.domain.name.com',
         hostedZone: someZone,
         certificate: Certificate.fromCertificateArn(this, 'someCert', certArn), // cert for elasticsearch.domain.name.com
 } 
```

What endups happening that I get a cname with: `elasticsearch.domain.name.com.domain.name.com`

Notice the duplication that happens.

I've validated that with AWS Support, I am using MonoCDK 1.132.0.

well it doesn't work becuase it needs to follow a regex that matches a domain.

So I tried do add maybe something like us-east-1.elasticsearch, but then it gets angery as the certificate didn't match :D

So basically the construct will only create if domainName is the actual domain name that matches the certificate and passess the regex check as a domain name.

however during creation of the recordset it joins itself with the hostedzone domain name making this monster up there.

I had to edit the zones manually after deployments sadly to mitigate that.

### Reproduction Steps

See above

### What did you expect to happen?

See above

### What actually happened?

See above

### CDK CLI Version

-

### Framework Version

1.132.0

### Node.js Version

-

### OS

-

### Language

Typescript

**Comment by mkianpour** [comment]
I had the same issue with python:
```
    "aws-cdk": "^2.54.0",
```
it joins domain name and zone name and it's not the right way because domain part is redundant

**Comment by camtauxe** [comment]
This bug is still present in Typescript CDK version 2.133. The issue should be re-opened.
